pex/scie/model.py

```diff
@@ -0,0 +1,250 @@
+
+    LINUX_AARCH64 = Value("linux-aarch64")
+    LINUX_X86_64 = Value("linux-x86_64")
+    MACOS_AARCH64 = Value("macos-aarch64")
+    MACOS_X86_64 = Value("macos-x86_64")
+    WINDOWS_AARCH64 = Value("windows-x86_64")
+    WINDOWS_X86_64 = Value("windows-aarch64")
+
+    @classmethod
+    def parse(cls, value):
+        # type: (str) -> SciePlatform.Value
+        return cls.current() if "current" == value else cls.for_value(value)
+
+    @classmethod
+    def current(cls):
+        # type: () -> SciePlatform.Value
+        system = platform.system().lower()
+        machine = platform.machine().lower()
+        if "linux" == system:
+            if machine in ("aarch64", "arm64"):
+                return cls.LINUX_AARCH64
+            elif machine in ("amd64", "x86_64"):
+                return cls.LINUX_X86_64
+        elif "darwin" == system:
+            if machine in ("aarch64", "arm64"):
+                return cls.MACOS_AARCH64
+            elif machine in ("amd64", "x86_64"):
+                return cls.MACOS_X86_64
+        elif "windows" == system:
+            if machine in ("aarch64", "arm64"):
+                return cls.WINDOWS_AARCH64
+            elif machine in ("amd64", "x86_64"):
+                return cls.WINDOWS_X86_64
+        raise ValueError(
+            "The current operating system / machine pair is not supported!: "
+            "{system} / {machine}".format(system=system, machine=machine)
+        )
+
+
+@attr.s(frozen=True)
+class ScieTarget(object):
+    platform = attr.ib()  # type: SciePlatform.Value
+    python_version = attr.ib()  # type: Union[Tuple[int, int], Tuple[int, int, int]]
+    pbs_release = attr.ib(default=None)  # type: Optional[str]
+
+    @property
+    def version_str(self):
+        # type: () -> str
+        return ".".join(map(str, self.python_version))
+
+
+@attr.s(frozen=True)
+class ScieInfo(object):
+    style = attr.ib()  # type: ScieStyle.Value
+    target = attr.ib()  # type: ScieTarget
+    file = attr.ib()  # type: str
+
+    @property
+    def platform(self):
+        # type: () -> SciePlatform.Value
+        return self.target.platform
+
+    @property
+    def python_version(self):
+        # type: () -> Union[Tuple[int, int], Tuple[int, int, int]]
+        return self.target.python_version
+
+
+@attr.s(frozen=True)
+class ScieOptions(object):
+    style = attr.ib(default=ScieStyle.LAZY)  # type: ScieStyle.Value
+    platforms = attr.ib(default=())  # type: Tuple[SciePlatform.Value, ...]
+    pbs_release = attr.ib(default=None)  # type: Optional[str]
+    python_version = attr.ib(
+        default=None
+    )  # type: Optional[Union[Tuple[int, int], Tuple[int, int, int]]]
+
+    def create_configuration(self, targets):
+        # type: (Targets) -> ScieConfiguration
+        return ScieConfiguration.from_targets(self, targets)
+
+
+@attr.s(frozen=True)
+class ScieConfiguration(object):
+    @classmethod
+    def from_tags(
```

Comment:
This is unused, but would / will be used by `pex3 scie create pre-existing.pex`. The idea is to get the wheel tags for all the distributions in the pre-existing PEX and use those to determine the platforms to target.